cache/purge.php

```diff
@@ -0,0 +1,50 @@
+<?php
+/**
+ *
+ * phpBB Media Embed PlugIn extension for the phpBB Forum Software package.
+ *
+ * @copyright (c) 2020 phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ */
+
+namespace phpbb\mediaembed\cache;
+
+use \Symfony\Component\Finder\Finder;
+
+/**
+ * mediaembed cron task.
+ */
+class purge
+{
+	/** @var \phpbb\cache\driver\driver_interface */
+	protected $cache;
+
+	/**
+	 * Constructor
+	 *
+	 * @param \phpbb\cache\driver\driver_interface  $cache      Cache jbject
+	 * @access public
+	 */
+	public function __construct(\phpbb\cache\driver\driver_interface $cache)
+	{
+		$this->cache = $cache;
+		$this->finder = new Finder();
+	}
+
+	/**
+	 * Purge all MediaEmbed cache files
+	 */
+	public function mediaembed_purge_cache()
+	{
+		$this->finder
+			->name('http.*')
+			->in($this->cache->cache_dir)
+			->files();
+
+		foreach ($this->finder as $file)
+		{
+			$this->cache->remove_file($file->getRealPath());
```

Comment:
What does `$file` look like? Can we instead use `cache->destroy($file)`?

Reply:
`$file` is an `SplFileInfo` object. Also things like `cache->destroy($file->getBasename())` don't work as `destroy()` method checks it against cache varnames and var_expires data which don't exist for s9e cache apparently.
Also `cache->remove_file()` is what `cache->destroy()` uses at the end anyway :)